src/Field/Definition/ChoiceField.php

```diff
@@ -93,6 +130,57 @@ public function validateData (ComponentContext $context, array $contentPath, mix
 		}
 	}
 
+
+	private function validateMultiSelect (ComponentContext $context, array $contentPath, mixed $data) : void
+	{
+		$context->ensureDataIsValid(
+			$contentPath,
+			$this,
+			$data,
+			[
+				new Type("array"),
+				new All([
+					new NotNull(),
+					new AtLeastOneOf([
+						new Type("string"),
+						new Type("int"),
+					]),
+				]),
+			],
+		);
+
+		\assert(\is_array($data));
```

Comment:
If it's empty: do we still get an array here?

Reply:
I've just tested it again: Optional multi-option fields will result in an missing value (it's completely missing from the JSON for some reason). So I've added an early exit for `null` values.